### PR TITLE
fix: using stale version of message callback for `useChannel` hook

### PR DESCRIPTION
**Pull request by ttypic**
Resolves https://github.com/ably-labs/react-hooks/issues/19
Jira links [SDK-3755]

[SDK-3755]: https://ably.atlassian.net/browse/SDK-3755?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/hooks/useChannel.test.tsx [hunk 1]
```diff
@@ -143,6 +142,63 @@ describe('useChannel', () => {
...
+                setCount((prev) => prev + 1);
+            });
+
+            return <div role="counter">{count}</div>;
+        };
+
+        renderInCtxProvider(ablyClient, <TestComponent />);
+
+        await act(async () => {
+            ablyClient.channels.get('blah').publish({ text: 'test message 1' });
```
Comment (ttypic) [inline src/hooks/useChannel.test.tsx:163]: I think we need to publish in different acts, to ensure state has been updated

Comment (stmoreau) [reply src/hooks/useChannel.test.tsx:163]: Done 😄

File: src/hooks/useChannel.test.tsx [hunk 2]
```diff
@@ -143,6 +142,63 @@ describe('useChannel', () => {
...
+
+    it('should use the latest version of the message callback', async () => {
+        let callbackCount = 0;
+
+        const TestComponent = () => {
+            const [count, setCount] = React.useState(0);
+
+            useChannel('blah', () => {
+                callbackCount++;
+                setCount((prev) => prev + 1);
```
[the remark this answers is not in the document]
Comment (stmoreau) [reply src/hooks/useChannel.test.tsx]: Updated 😄

**Review by stmoreau — APPROVED** [review]

**Comment by jlil** [comment]
hi - is the v3 package published anywhere? thanks!

**Comment by ttypic** [comment]
Hi @jlil,

Yes, we have canary build, you can try it `npm i @ably-labs/react-hooks@3.0.0-canary.1`. 

**Disclaimer:** it's highly experimental right now, and has number of breaking changes, documentation to it is in the [integration/v3](https://github.com/ably-labs/react-hooks/tree/integration/v3) branch.

**Pull request merged**